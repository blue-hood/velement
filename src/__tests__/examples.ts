import '@testing-library/jest-dom/extend-expect';
import VirtualElement, { appendChildren, createElement } from '..';

test('Usage with JavaScript', async () => {
  class Div extends VirtualElement {
    // eslint-disable-next-line @typescript-eslint/explicit-member-accessibility
    constructor(element) {
      super(element || 'div');
      this.element.innerHTML = 'VirtualDivElement';
    }
  }

  const container = document.createElement('div');

  const htmlElement = createElement('div', null);
  htmlElement.innerHTML = `HTMLDivElement`;
  const virtualElement = createElement(Div, {});

  appendChildren(container, 'TextNode', htmlElement, virtualElement);
  expect(container).toMatchSnapshot();
});

test('Usage with TypeScript', async () => {
  class Div extends VirtualElement<HTMLDivElement> {
    public constructor(element: HTMLDivElement | null) {
      super(element || 'div');
      this.element.innerHTML = 'VirtualDivElement';
    }
  }

  const container = document.createElement('div');

  const htmlElement = createElement<HTMLDivElement>('div', null);
  htmlElement.innerHTML = `HTMLDivElement`;
  const virtualElement = createElement<Div, {}>(Div, {});

  appendChildren(container, 'TextNode', htmlElement, virtualElement);
  expect(container).toMatchSnapshot();
});
