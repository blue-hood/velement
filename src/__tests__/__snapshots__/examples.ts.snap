// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Usage with JavaScript 1`] = `
<div>
  TextNode
  <div>
    HTMLDivElement
  </div>
  <div>
    VirtualDivElement
  </div>
</div>
`;

exports[`Usage with TypeScript 1`] = `
<div>
  TextNode
  <div>
    HTMLDivElement
  </div>
  <div>
    VirtualDivElement
  </div>
</div>
`;
